File: java-client/src/SolvexClient.java
package com.solvexai;

import java.net.HttpURLConnection;
import java.net.URL;
import java.util.Scanner;

public class SolvexClient {
    private String apiKey;
    private final String BASE_URL = "https://api.solvex.ai";

    public SolvexClient(String apiKey) {
        this.apiKey = apiKey;
    }

    public String getLeaderboard() throws Exception {
        URL url = new URL(BASE_URL + "/leaderboard");
        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
        conn.setRequestMethod("GET");
        conn.setRequestProperty("Authorization", "Bearer " + apiKey);

        Scanner scanner = new Scanner(conn.getInputStream());
        String response = scanner.useDelimiter("\\A").next();
        scanner.close();
        return response;
    }
}

File: php-client/src/SolvexClient.php
<?php

class SolvexClient {
    private $apiKey;
    private $baseUrl = "https://api.solvex.ai";

    public function __construct($apiKey) {
        $this->apiKey = $apiKey;
    }

    public function getLeaderboard() {
        $ch = curl_init("{$this->baseUrl}/leaderboard");
        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
        curl_setopt($ch, CURLOPT_HTTPHEADER, ["Authorization: Bearer {$this->apiKey}"]);
        $response = curl_exec($ch);
        curl_close($ch);
        return $response;
    }
}
?>
